apollo-client/src/main/java/com/ctrip/framework/apollo/spring/config/PropertySourcesProcessor.java

```diff
@@ -60,7 +67,25 @@ protected void initializePropertySources() {
       int order = iterator.next();
       for (String namespace : NAMESPACE_NAMES.get(order)) {
         Config config = ConfigService.getConfig(namespace);
-
+        config.addChangeListener(new ConfigChangeListener() {
+          @Override
+          public void onChange(ConfigChangeEvent changeEvent) {
+            Set<String> keys = changeEvent.changedKeys();
+            if (CollectionUtils.isEmpty(keys)){
+              return;
+            }
+            for (String k:keys){
+              ConfigChange configChange = changeEvent.getChange(k);
+              Collection<SpringValue> targetValues = ApolloAnnotationProcessor.monitor().get(k);
```

Comment:
If we have a class called `SpringValueRegistry`, then we could simply pass the changed key to it and let it do the update logic.
And do remember NOT to pass the new value, because we should always get the value from Spring `environment` using environment.getProperty() since the key might be overriden in a different hierarchy.

Reply:
I'll think about this carefully.